src/stream.rs

```diff
@@ -60,6 +60,11 @@ impl<R: BufRead> Read for Deflate64Decoder<R> {
                 ));
             }
 
+            if buf.is_empty() {
+                // we received empty buffer, so it won't be possible to write anything
+                return Ok(0);
+            }
```

Comment:
Shall we return this at the start of the function?

I don't see why we should do anything or return an error when the buffer is empty?

Reply:
Other implementation like flate2 tries to consume data from base Read implementation so I chose afrer reading from base stream but I think it's ok to early return at the top.